pkg/api/grpc/subscribe.go

```diff
@@ -82,17 +81,18 @@ func (a *api) streamSubscribe(stream runtimev1pb.Dapr_SubscribeTopicEventsAlpha1
 			Routes:          subapi.Routes{Default: "/"},
 		},
 	})
+	if err != nil {
+		return err
+	}
 
-	if err := a.processor.Subscriber().ReloadPubSub(req.GetPubsubName()); err != nil {
+	if err := a.processor.Subscriber().StartStreamerSubscription(key); err != nil {
```

Comment:
Can key ever be empty or nil and if so should we do a nil check before starting the streamer subscription? I only see `key` being returned here `key := a.pubsubAdapterStreamer.StreamerKey(req.GetPubsubName(), req.GetTopic())` but am uncertain if it could cause issues if returned unset and used?

Reply:
No, key is just a string concatenation function of the given strings so it won't be nil or empty https://github.com/dapr/dapr/blob/47a5b3a1762ca7e78e6921333e206f520b1e6d02/pkg/runtime/pubsub/streamer/streamer.go#L167